connection.go

```diff
@@ -144,6 +144,10 @@ func (mc *mysqlConn) interpolateParams(query string, args []driver.Value) (strin
 	buf = buf[:0]
 	argPos := 0
 
+	if len(args) != strings.Count(query, "?") {
+		return "", ErrArgsInvalid
```

Comment:
@methane 
Only when interploateParams=true, db.Query and db.Exec will try client-side prepare